src/test/java/com/algorand/algosdk/integration/Applications.java

```diff
@@ -247,4 +254,105 @@ public void checkAccountData(
 
         assertThat(found).as("Couldn't find key '%s'", hasKey).isTrue();
     }
+
+    @Then("according to {string}, the contents of the box with name {string} in the current application should be {string}. If there is an error it is {string}.")
+    public void contentsOfBoxShouldBe(String fromClient, String encodedBoxName, String boxContents, String errStr) throws Exception {
+        Response<Box> boxResp;
+        if (fromClient.equals("algod"))
+            boxResp = clients.v2Client.GetApplicationBoxByName(this.appId).name(encodedBoxName).execute();
+        else if (fromClient.equals("indexer"))
+            boxResp = clients.v2IndexerClient.lookupApplicationBoxByIDandName(this.appId).name(encodedBoxName).execute();
+        else
+            throw new IllegalArgumentException("expecting algod or indexer, got " + fromClient);
+
+        // If an error was expected, make sure it is set correctly.
+        if (StringUtils.isNotEmpty(errStr)) {
+            assertThat(boxResp.isSuccessful()).isFalse();
+            assertThat(boxResp.message()).containsIgnoringCase(errStr);
+            return;
+        }
+
+        assertThat(boxResp.body().value()).isEqualTo(boxContents);
+    }
+
+    private static boolean contains(byte[] elem, List<byte[]> xs) {
+        for (byte[] e : xs) {
+            if (Arrays.equals(e, elem))
+                return true;
+        }
+        return false;
+    }
+
+    @Then("according to {string}, the current application should have the following boxes {string}.")
+    public void checkAppBoxes(String fromClient, String encodedBoxesRaw) throws Exception {
+        Response<BoxesResponse> r;
+        if (fromClient.equals("algod"))
+            r = clients.v2Client.GetApplicationBoxes(this.appId).execute();
+        else if (fromClient.equals("indexer"))
+            r = clients.v2IndexerClient.searchForApplicationBoxes(this.appId).execute();
+        else
+            throw new IllegalArgumentException("expecting algod or indexer, got " + fromClient);
+
+        Assert.assertTrue(r.isSuccessful());
+
+        final List<byte[]> expectedNames = Lists.newArrayList();
+        if (!encodedBoxesRaw.isEmpty()) {
+            for (String s : Strings.split(encodedBoxesRaw, ':')) {
+                expectedNames.add(Encoder.decodeFromBase64(s));
+            }
+        }
+
+        final List<byte[]> actualNames = Lists.newArrayList();
+        for (BoxDescriptor b : r.body().boxes) {
+            actualNames.add(b.name);
+        }
+
+        Assert.assertEquals("expected and actual box names length do not match", expectedNames.size(), actualNames.size());
+        for (byte[] e : expectedNames) {
+            if (!contains(e, actualNames))
+                throw new RuntimeException("expected and actual box names do not match: " + expectedNames + " != " + actualNames);
```

Comment:
A less error-prone way to compare lists would be to sort and directly compare their contents